api/src/main/java/org/xwiki/apps/googleapps/internal/GoogleAppsManagerImpl.java

```diff
@@ -0,0 +1,1372 @@
+     * */
+    @Unstable
+    public boolean isActive(XWikiContext context)
+    {
+        if (configActiveFlag == null) {
+            readConfigDoc(context);
+        }
+        if (authService == null) {
+            try {
+                initialize();
+            } catch (InitializationException e) {
+                e.printStackTrace();
+            }
+        }
+        if (configActiveFlag != null) {
+            return configActiveFlag;
+        }
+        return false;
+    }
+
+    /**
+     * @return true if the app is configured to use cookies to store the association to the Google user.
+     * @since 3.0
+     */
+    @Unstable
+    public boolean isUseCookies() {
+        return useCookies;
+    }
+
+    /**
+     * @return true if the app is configured to simply use the cookie and thus recognize the user based on cookie
+     * @since 3.0
+     */
+    @Unstable
+    public boolean isSkipLoginPage() {
+        return skipLoginPage;
+    }
+
+    /**
+     * @return true if the app is configured to use cookies to store the association to the Google user
+     * @since 3.0
+     */
+    @Unstable
+    public boolean isAuthWithCookies() {
+        return authWithCookies;
+    }
+
+    /**
+     * @return how long (in seconds) the cookies should be valid
+     * @since 3.0
+     */
+    @Unstable
+    long getConfigCookiesTTL() {
+        return configCookiesTTL;
+    }
+
+    private void readConfigDoc(XWikiContext context) {
+
+        try {
+            log.warn("Attempting to fetch Config doc");
+            BaseObject config = getConfigDoc(context);
+            if (config != null) {
+                configActiveFlag   = 0 != config.getIntValue("activate");
+                useCookies         = 0 != config.getIntValue("useCookies");
+                skipLoginPage      = 0 != config.getIntValue("skipLoginPage");
+                authWithCookies    = 0 != config.getIntValue("authWithCookies");
+                configAppName      = config.getStringValue("appname").trim();
+                configClientId     = config.getStringValue("clientid").trim();
+                configClientSecret = config.getStringValue("secret").trim();
+                configDomain       = config.getStringValue("domain").trim();
+                if (configDomain.length() == 0) {
+                    configDomain = null;
+                }
+                List<String> configScopes = Arrays.asList(config.getStringValue("scope").split("\\s"));
+                configScopeUseAvatar = configScopes.contains(AVATAR);
+                configScopeUseDrive = configScopes.contains("drive");
+                configCookiesTTL = config.getLongValue("cookiesTTL");
+            }
+        } catch (XWikiException e) {
+            e.printStackTrace();
+            if (log != null) {
+                log.warn("can't fetch Config doc");
+            }
+        }
+    }
+
+    /**
+     * Reads the manifest to find when the JAR file was assembled by maven.
```

Comment:
I don't really understand what's the purpose of this need.

Reply:
This supports testing and allows a wiki-page to make sure that the last version was properly deployed to the wiki.

Reply:
I don't think this kind of mechanism was ever used before in XWiki. There is the extension's version and I think that's enough. WDYT @mflorea?

Reply:
No, the extension's version is not enough to test if the snapshot is the last built.
paul

Reply:
Why do you need to know this? You have your XWiki instance and install  the application and the current installed version should work.

Reply:
Well, this was to compensate frustrating experiences where, in particular, the configuration of a local m2 repository was not completely documented. I sure can remove it. I find this useful and clearly this is not specific to GoogleApps.

Reply:
I've now included the result of this method and a method "getBuildVersion" as the last line of GoogleApps/WebHome, probably the best place for an "about" page.